CMakeLists.txt

```diff
@@ -24,6 +24,13 @@ project(${PACKAGE_NAME} CXX C)
 
 set(PROJECT_THIRD_PARTY_DIR "${PROJECT_SOURCE_DIR}/third_party")
 
+# WARNING: do not change any compiler flags or variables in this
+# CMakeList file.  The grpc subdirectory (and its childer), have
+# carefully crafted flags, and you may break them by using your own
+# settings.  In particular -Werror -Wall breaks protobuf or grpc in many
```

Comment:
That's sad and unfortunate. Do you think it would be worthwhile to raise an issue for those teams to fix their code, or is there already an existing issue we can reference here?

Reply:
That is a deep rabbit hole.  There are many such bugs, e.g. grpc/grpc#11376, which references bugs in openssl, I think even zlib breaks with gcc-7.x.  I got other problems with clang-3.9 and higher, the code generates warnings about -Wexpansion-to-defined.  That one we should file a bug about because I cannot find anything.